pkg/kubelet/kubelet_pods.go

```diff
@@ -1630,7 +1630,21 @@ func (kl *Kubelet) determinePodResizeStatus(pod *v1.Pod, podStatus *v1.PodStatus
 	specStatusDiffer := false
 	for _, c := range pod.Spec.Containers {
 		if cs, ok := podutil.GetContainerStatus(podStatus.ContainerStatuses, c.Name); ok {
-			if cs.Resources != nil && !cmp.Equal(c.Resources, *cs.Resources) {
+			cResourceCopy := c.Resources.DeepCopy()
+			csResourceCopy := cs.Resources.DeepCopy()
+			// for both requests and limits, we only compare the cpu, memory and ephemeralstorage
+			// which are included in convertToAPIContainerStatuses
+			for resource := range cResourceCopy.Limits {
+				if resource != v1.ResourceCPU && resource != v1.ResourceMemory && resource != v1.ResourceEphemeralStorage {
```

Comment:
extract a small method like `isResourceVPAAllowed `etc since it can be long in future.

Reply:
Did we test emphemeral storage?